core/class/script.class.php

```diff
@@ -229,13 +229,14 @@ public function execute($_options = null) {
 					return script::$_requet_cache[$request];
 				}
 				$cmd = 'sudo chmod +x ' . explode(' ', $request)[0] . ' 2>/dev/null;';
-				if (strpos($request, '.php') !== false) {
+				$force_interpreter = $this->getConfiguration('doNotForceInterpreter', '0') == '0';
```

Comment:
Ça me gène quand on nomme quelque chose avec "doNot". On peut très vite partir sur des doubles négations et ça deviens très compliqué d’interpréter correctement la règle humainement.

exemple : `$notOk !== false`

Je serai partisant d’une option `forceInterpreter`.

```suggestion
				$force_interpreter = $this->getConfiguration('forceInterpreter', '1') != '0';
```

Comme ça on garde la même logique et la rétro-compatibilité.

Reply:
Je dois avouer avoir hésité, je fais les corrections

edit: pour éviter de forcer l'option 'forceInterpreter' à 1 et donc de cocher la case par défaut, ce qui peut perturber les utilisateurs, je baptise le bouton 'useShebang'. Comme ça la case reste décochée, le test est simplement inversé : 
```php
$use_shebang = $this->getConfiguration('useShebang', '0') == '1';
if (!$use_shebang && strpos($request, '.php') !== false) {
	$cmd .= 'php ' . $request;
} elseif (!$use_shebang && strpos($request, '.rb') !== false) {
	$cmd .= 'ruby ' . $request;
} elseif (!$use_shebang && strpos($request, '.py') !== false) {
	$cmd .= 'python ' . $request;
} elseif (!$use_shebang && strpos($request, '.pl') !== false) {
	$cmd .= 'perl ' . $request;
} else {
	$cmd .= $request;
}
```